sklearn/linear_model/tests/test_ridge.py

```diff
@@ -1008,26 +1008,50 @@ def test_n_iter():
         assert reg.n_iter_ is None
 
 
-def test_ridge_fit_intercept_sparse():
-    X, y = make_regression(n_samples=1000, n_features=2, n_informative=2,
-                           bias=10., random_state=42)
+@pytest.mark.parametrize('solver', ['sparse_cg', 'auto'])
+def test_ridge_fit_intercept_sparse(solver):
+    X, y = _make_sparse_offset_regression(n_features=20, random_state=0)
+    X_csr = sp.csr_matrix(X)
+
+    # for now only sparse_cg can fit an intercept with sparse X with default
+    # tol and max_iter, sag should raise a warning and is handled in
+    # test_ridge_fit_intercept_sparse_sag
+    # "auto" should switch to "sparse_cg"
+    dense_ridge = Ridge(alpha=1., solver='sparse_cg', fit_intercept=True)
```

Comment:
sparse_cg for dense_ridge?

Reply:
sparse_cg can fit both sparse and dense data. since both "auto" and "sparse_cg" should result in "sparse_cg" being used when X is sparse, the reference is "sparse_cg" with dense data, and Ridge(solver="auto") and Ridge(solver="sparse_cg"), fitted on sparse data, are compared to it